### PR TITLE
imp: ics20 v2 self review part 3

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/ibc_module.go [hunk 1]
```diff
@@ -240,26 +240,28 @@ func (im IBCModule) OnRecvPacket(
...
 		eventAttributes = append(eventAttributes, sdk.NewAttribute(types.AttributeKeyAckError, ackErr.Error()))
 	}
 
-	ctx.EventManager().EmitEvent(sdk.NewEvent(
-		sdk.EventTypeMessage,
-		eventAttributes...,
-	))
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
```
Comment (damiannolan) [inline modules/apps/transfer/ibc_module.go:257]: Should this be `EventTypePacket`?

File: modules/apps/transfer/keeper/msg_server.go [hunk 2]
```diff
@@ -38,10 +38,8 @@ func (k Keeper) Transfer(goCtx context.Context, msg *types.MsgTransfer) (*types.
 		return nil, errorsmod.Wrapf(ibcerrors.ErrInvalidRequest, "cannot transfer multiple coins with ics20-1")
 	}
 
-	for _, coin := range coins {
-		if !k.bankKeeper.IsSendEnabledCoin(ctx, coin) {
-			return nil, errorsmod.Wrapf(types.ErrSendDisabled, "transfers are currently disabled for %s", coin.Denom)
-		}
+	if err := k.bankKeeper.IsSendEnabledCoins(ctx, coins...); err != nil {
```
Comment (damiannolan) [inline modules/apps/transfer/keeper/msg_server.go:41]: much nicer

**Review by damiannolan — APPROVED** [review]
Nice improvements! :)

File: modules/apps/transfer/ibc_module.go (hunk 1 above)
Comment (colin-axner) [reply modules/apps/transfer/ibc_module.go:257]: Oh yes it should

File: modules/apps/transfer/ibc_module.go [hunk 3]
```diff
@@ -240,26 +240,28 @@ func (im IBCModule) OnRecvPacket(
...
 		eventAttributes = append(eventAttributes, sdk.NewAttribute(types.AttributeKeyAckError, ackErr.Error()))
 	}
 
-	ctx.EventManager().EmitEvent(sdk.NewEvent(
-		sdk.EventTypeMessage,
-		eventAttributes...,
-	))
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypePacket,
```
Comment (colin-axner) [inline modules/apps/transfer/ibc_module.go:257]:
```suggestion
			types.EventTypePacket,
```

**Pull request merged**